c2rust-analyze/src/analyze.rs

```diff
@@ -664,19 +660,19 @@ fn run(tcx: TyCtxt) {
             pointee_type::generate_constraints(&acx, &mir)
         }));
 
-        let pointee_constraints = match r {
-            Ok(x) => x,
+        let mut info = FuncInfo::default();
+        let local_pointee_types = LocalPointerTable::new(acx.num_pointers());
+        info.acx_data.set(acx.into_data());
+
+        match r {
+            Ok(pointee_constraints) => {
+                info.pointee_constraints.set(pointee_constraints);
+            }
             Err(pd) => {
                 gacx.mark_fn_failed(ldid.to_def_id(), pd);
-                continue;
             }
```

Comment:
Was this previously jumping back to line 623? Why isn't the `continue` needed anymore?

Reply:
Looks like this change is part of 6d7d3e3dfc253577c25a240888d842d04edc171c "preserve function acx_data on error".  This loop is the one that initializes the `FuncInfo` (which contains `acx_data`) for each function.  Removing the `continue` means that if an error occurs in this part of the analysis, only `info.pointee_constraints` will be affected, and all other `info` fields will still get initialized.